main.go

```diff
@@ -2,14 +2,59 @@ package main
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"io"
 	"os"
 
 	"github.com/databricks/cli/cmd"
 	"github.com/databricks/cli/cmd/root"
+	"github.com/databricks/cli/libs/telemetry"
+	"github.com/databricks/databricks-sdk-go/logger"
 )
 
 func main() {
 	ctx := context.Background()
+
+	// A new process is spawned for uploading telemetry data. We handle this separately
+	// from the rest of the CLI commands.
+	// This is done because [root.Execute] spawns a new process to run the
+	// "telemetry upload" command if there are logs to be uploaded. Having this outside
+	// of [root.Execute] ensures that the telemetry upload process is not spawned
+	// infinitely in a recursive manner.
+	if len(os.Args) == 3 && os.Args[1] == "telemetry" && os.Args[2] == "upload" {
+		var err error
+
+		// By default, this command should not write anything to stdout or stderr.
+		outW := io.Discard
+		errW := io.Discard
+
+		// If the environment variable is set, redirect stdout to the file.
+		// This is useful for testing.
+		if v := os.Getenv(telemetry.UploadLogsFileEnvVar); v != "" {
+			outW, _ = os.OpenFile(v, os.O_CREATE|os.O_WRONLY, 0o644)
+			errW = outW
```

Comment:
what about CLI and SDK logging, are they also writing to this file?

Reply:
Thanks, actually this is no longer required since the the daemon library takes care of redirecting the stdout / stderr to the log file here:
```
	d := daemon.Daemon{
		Args:        []string{"telemetry", "upload"},
		Env:         env,
		PidFilePath: os.Getenv(telemetry.PidFileEnvVar),
		LogFile:     os.Getenv(telemetry.UploadLogsFileEnvVar),
	}
```

Note that if LogFile is `""` then the stdout / stderr is redirected to `/dev/null`.

> what about CLI and SDK logging

The would directly write to stdout / stderr. That's why to avoid the INFO / WARN logs from the SDK leaking to the output I added: 
```
// Suppress non-error logs from the SDK.
logger.DefaultLogger = &logger.SimpleLogger{
	Level: logger.LevelError,
}
```